pkg/addon/controllers/addonconfiguration/graph.go

```diff
@@ -369,16 +369,25 @@ func (n *installStrategyNode) addNode(addon *addonv1alpha1.ManagedClusterAddOn)
 		// TODO we should also filter out the configs which are not supported configs.
 		overrideConfigMapByAddOnConfigs(n.children[addon.Namespace].desiredConfigs, addon.Spec.Configs)
 
-		// copy the spechash from mca status
-		for _, configRef := range addon.Status.ConfigReferences {
-			if configRef.DesiredConfig == nil {
-				continue
-			}
-			if nodeDesiredConfigArray, ok := n.children[addon.Namespace].desiredConfigs[configRef.ConfigGroupResource]; ok {
-				for i, nodeDesiredConfig := range nodeDesiredConfigArray {
-					if nodeDesiredConfig.DesiredConfig.ConfigReferent == configRef.DesiredConfig.ConfigReferent {
-						n.children[addon.Namespace].desiredConfigs[configRef.ConfigGroupResource][i].DesiredConfig.SpecHash = configRef.DesiredConfig.SpecHash
+		//	go through mca spec configs and copy the specHash from status if they match
+		for _, config := range addon.Spec.Configs {
+			for _, configRef := range addon.Status.ConfigReferences {
+				if configRef.DesiredConfig == nil {
+					continue
+				}
+
+				// compare the ConfigGroupResource and ConfigReferent
+				if configRef.ConfigGroupResource == config.ConfigGroupResource &&
```

Comment:
we commonly want to avoid deep nested code. Would be something like this ok?

```
if configRef.ConfigGroupResource != config.ConfigGroupResource || configRef.DesiredConfig.ConfigReferent != config.ConfigReferent {
continue
}
nodeDesiredConfigArray, ok := n.children[addon.Namespace].desiredConfigs[configRef.ConfigGroupResource]
if !ok {
continue
}

for i, nodeDesiredConfig := range nodeDesiredConfigArray {
							if nodeDesiredConfig.DesiredConfig.ConfigReferent == configRef.DesiredConfig.ConfigReferent {
								n.children[addon.Namespace].desiredConfigs[configRef.ConfigGroupResource][i].DesiredConfig.SpecHash = configRef.DesiredConfig.SpecHash
							}
						}

```

Reply:
fixed